### ISSUE TITLE
Needs an API

### ISSUE THREAD

**Issue by mattstauffer** [mattstauffer/pulledover#17]
We have some potential iOS devs, but we need an API.

OAuth 2.0 definitely. https://github.com/lucadegasperi/oauth2-server-laravel (see [Symposium](http://github.com/tightenco/symposium) for a sample implementation)

I think I'd be OK with it being the same codebase, just at pulledover.us/api/

Some resources, I guess:

/friends
/numbers

I'm curious whether it even merits using Twilio for the mobile app. I wonder if the mobile version should use S3 for recordings? If so, recordings probably need to mature a bit in our system to handle more than just twilio recordings.

But that's interesting--would have to make sure we can stream so that, if someone smashed your phone mid recording, it didn't all get lost.

I guess there are two components. One is opening up an API to manage everything other than recordings; two is allowing an app to handle recordings, likely not via Twilio, but maybe? (Either way we're definitely going to keep it working via web and Twilio, because I highly value non-smartphone-users having an option)

**Comment by RDelorier** [comment]
I can definitely help out with the api side, ill start off with the oauth setup. Any plans on moving to L5.2 with the multiple auth drivers?

**Comment by mattstauffer** [comment]
Yes sir! Was gonna do a Laravel shift upgrade in th next day or two. Maybe
I do it this morning?

**Comment by RDelorier** [comment]
Sweet, If you go ahead and switch it then I'll start working on this tonight. Thanks for the link to Symposium that should make it pretty easy to figure out how to setup the oath package.

**Comment by matthewtrask** [comment]
yooo, still need some endpoints built?

**Comment by mattstauffer** [comment]
@matthewtrask Thanks dude! Honestly, interest hasn't stayed very high. You can do it for fun, or maybe building this might make it possible to build an app and then that can push up the interest level.. I just haven't had any time to work on this in quite a while.
